[Crash] App crashes after connection

Hey, every time after successful connection to MA server, app crashes. Crash logs say this: 
```1775468872144 Debug/WebRTCTransport: Created new WebRTC manager [129359708]
1775468872144 Info/WebRTCConnectionManager: Starting WebRTC connection
1775468872144 Info/SignalingClient: Connecting to signaling server: [REDACTED_URL]
1775468872164 Info/MainDataSource: SessionState changed: Connecting
1775468872164 Info/MainDataSource: Connecting - stopping Sendspin
1775468872164 Info/SendspinClientFactory: Destroying shared audio pipeline
1775468872225 Info/ServiceClient: External consumer active
1775468872238 Info/MainMediaPlaybackService: Registered audio device callback for routing change detection
1775468872365 Info/SignalingClient: Connected to signaling server
1775468872366 Debug/WebRTCConnectionManager: Sending ConnectRequest message
1775468872368 Debug/SignalingClient: Sending signaling message: connect-request
1775468872422 Debug/SignalingClient: Received signaling message
1775468872427 Debug/SignalingClient: Parsed signaling message type: connected
1775468872428 Debug/WebRTCConnectionManager: Received signaling message: connected
1775468872428 Info/WebRTCConnectionManager: Connected. ICE servers: 4
1775468872429 Info/PeerConnectionWrapper[Android]: Initializing peer connection with 4 ICE servers
1775468872493 Info/ServiceClient: App foregrounded
1775468872533 Debug/PeerConnectionWrapper[Android]: Peer connection initialized
1775468872533 Debug/WebRTCConnectionManager: Peer connection state: new
1775468872534 Debug/WebRTCConnectionManager: Creating ma-api data channel
1775468872534 Debug/PeerConnectionWrapper[Android]: Creating data channel: ma-api (ordered=true, maxRetransmits=-1)
1775468872537 Debug/WebRTCConnectionManager: Creating sendspin data channel
1775468872537 Debug/PeerConnectionWrapper[Android]: Creating data channel: sendspin (ordered=true, maxRetransmits=-1)
1775468872539 Debug/WebRTCConnectionManager: Creating SDP offer
1775468872539 Debug/PeerConnectionWrapper[Android]: Creating SDP offer
1775468872539 Debug/WebRTCConnectionManager: Sendspin data channel state: Connecting
1775468872554 Debug/PeerConnectionWrapper[Android]: SDP offer created and set as local description
1775468872555 Debug/WebRTCConnectionManager: Sending SDP offer
1775468872558 Debug/SignalingClient: Sending signaling message: offer
1775468872579 Info/MainMediaPlaybackService: Unregistered audio device callback
1775468872610 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775468872611 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775468872612 Debug/SignalingClient: Sending signaling message: ice-candidate
1775468872613 Info/AutoLibrary: Items for auto_lib_root
1775468872614 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775468872615 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775468872616 Debug/SignalingClient: Sending signaling message: ice-candidate
1775468872633 Info/ServiceClient: External consumer inactive
1775468872674 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775468872675 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775468872675 Debug/SignalingClient: Sending signaling message: ice-candidate
1775468877903 Debug/SignalingClient: Received signaling message
1775468877905 Debug/PeerConnectionWrapper[Android]: Connection state: connecting
1775468877906 Debug/WebRTCConnectionManager: Peer connection state: connecting
1775468877907 Debug/SignalingClient: Parsed signaling message type: answer
1775468877908 Debug/WebRTCConnectionManager: Received signaling message: answer
1775468877908 Info/WebRTCConnectionManager: Received SDP answer
1775468877908 Debug/PeerConnectionWrapper[Android]: Setting remote answer
1775468877917 Debug/PeerConnectionWrapper[Android]: Remote answer set successfully
1775468877917 Debug/WebRTCConnectionManager: Remote answer set successfully
1775468877985 Debug/PeerConnectionWrapper[Android]: Connection state: connected
1775468877988 Debug/WebRTCConnectionManager: Peer connection state: connected
1775468878031 Debug/WebRTCConnectionManager: Sendspin data channel state: Open
1775468878032 Info/WebRTCConnectionManager: Sendspin data channel ready for use
1775468878035 Info/MainDataSource: SessionState changed: WebRTC
1775468878036 Info/MainDataSource: Clearing all cached data
1775468878098 Debug/AuthVM: SessionState changed: WebRTC
1775468878098 Debug/AuthVM: DataConnectionState: AwaitingServerInfo
1775468878139 Info/MainDataSource: SessionState changed: WebRTC
1775468878140 Info/MainDataSource: Clearing all cached data
1775468878205 Info/MainDataSource: SessionState changed: WebRTC
1775468878206 Info/MainDataSource: Clearing all cached data
1775468878206 Debug/AuthVM: SessionState changed: WebRTC
1775468878206 Debug/AuthVM: DataConnectionState: AwaitingAuth
1775468878206 Debug/AuthVM: AwaitingAuth - checking auth process state
1775468878206 Debug/AuthVM: Calling loadProviders()
1775468878206 Debug/AuthVM: loadProviders() called, current providers count: 0
1775468878206 Debug/AuthVM: WebRTC connection - using builtin provider directly (skip API call)
1775468878487 Debug/ServiceClient: Saved token for server
1775468878487 Info/MainDataSource: SessionState changed: WebRTC
1775468878487 Debug/AuthVM: SessionState changed: WebRTC
1775468878488 Debug/AuthVM: DataConnectionState: Authenticated
1775468878491 Warn/MainDataSource: Cannot create Sendspin client: Sendspin disabled in settings
1775468878491 Info/MainDataSource: Updating players and queues
1775468879173 Info/MainMediaPlaybackService: Registered audio device callback for routing change detection
1775468879849 Info/ServiceClient: App backgrounded
1775468879915 Info/ServiceClient: External consumer active

=== CRASH ===
java.lang.IllegalArgumentException: Software rendering doesn't support hardware bitmaps
	at android.graphics.BaseCanvas.throwIfHwBitmapInSwMode(BaseCanvas.java:732)
	at android.graphics.BaseCanvas.throwIfCannotDraw(BaseCanvas.java:82)
	at android.graphics.BaseCanvas.drawBitmap(BaseCanvas.java:166)
	at android.graphics.Canvas.drawBitmap(Canvas.java:1640)
	at android.graphics.Bitmap.createScaledAshmemBitmap(Bitmap.java:989)
	at android.media.MediaMetadata$Builder.scaleBitmap(MediaMetadata.java:1019)
	at android.media.MediaMetadata$Builder.build(MediaMetadata.java:1002)
	at android.media.session.MediaSession.setMetadata(MediaSession.java:613)
	at android.support.v4.media.session.MediaSessionCompat$MediaSessionImplApi21.setMetadata(MediaSessionCompat.java:4100)
	at android.support.v4.media.session.MediaSessionCompat.setMetadata(MediaSessionCompat.java:799)
	at io.music_assistant.client.services.SharedMediaSessionManager.writePlaybackToSession(SharedMediaSessionManager.kt:229)
	at io.music_assistant.client.services.SharedMediaSessionManager.clearErrorState(SharedMediaSessionManager.kt:116)
	at io.music_assistant.client.services.AndroidAutoPlaybackService$observeSessionState$1$1.emit(AndroidAutoPlaybackService.kt:265)
	at io.music_assistant.client.services.AndroidAutoPlaybackService$observeSessionState$1$1.emit(AndroidAutoPlaybackService.kt:261)
	at kotlinx.coroutines.flow.StateFlowImpl.collect(StateFlow.kt:401)
	at kotlinx.coroutines.flow.ReadonlyStateFlow.collect(Share.kt:2)
	at io.music_assistant.client.services.AndroidAutoPlaybackService$observeSessionState$1.invokeSuspend(AndroidAutoPlaybackService.kt:261)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:124)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:820)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@14c6f27, Dispatchers.IO]
```

I tried connecting via webrtc and url. Problems seems with screen rendering.

EDIT: forgot to mention I'm using MA server 2.9.0 beta 5, MA app 2026.04.04 debug release

edit2: markdown fix

When I try to go to app settings before it crashes, it doesn't crash, and works normally after. You have to tap the screen where the gear icon is. When you exit and go back in, it crashes again. 
```
1775470497353 Debug/WebRTCTransport: Created new WebRTC manager [246518727]
1775470497353 Info/WebRTCConnectionManager: Starting WebRTC connection
1775470497353 Info/SignalingClient: Connecting to signaling server: [REDACTED_URL]
1775470497360 Info/MainDataSource: SessionState changed: ByUser
1775470497360 Info/MainDataSource: Disconnected by user - clearing all data
1775470497361 Info/SendspinClientFactory: Destroying shared audio pipeline
1775470497361 Info/MainDataSource: Clearing all cached data
1775470497600 Info/SignalingClient: Connected to signaling server
1775470497601 Debug/WebRTCConnectionManager: Sending ConnectRequest message
1775470497603 Debug/SignalingClient: Sending signaling message: connect-request
1775470497658 Debug/SignalingClient: Received signaling message
1775470497663 Debug/SignalingClient: Parsed signaling message type: connected
1775470497663 Debug/WebRTCConnectionManager: Received signaling message: connected
1775470497663 Info/WebRTCConnectionManager: Connected. ICE servers: 4
1775470497665 Info/PeerConnectionWrapper[Android]: Initializing peer connection with 4 ICE servers
1775470497738 Info/ServiceClient: App foregrounded
1775470497780 Debug/PeerConnectionWrapper[Android]: Peer connection initialized
1775470497780 Debug/WebRTCConnectionManager: Creating ma-api data channel
1775470497780 Debug/PeerConnectionWrapper[Android]: Creating data channel: ma-api (ordered=true, maxRetransmits=-1)
1775470497781 Debug/WebRTCConnectionManager: Peer connection state: new
1775470497785 Debug/WebRTCConnectionManager: Creating sendspin data channel
1775470497785 Debug/PeerConnectionWrapper[Android]: Creating data channel: sendspin (ordered=true, maxRetransmits=-1)
1775470497787 Debug/WebRTCConnectionManager: Creating SDP offer
1775470497787 Debug/PeerConnectionWrapper[Android]: Creating SDP offer
1775470497787 Debug/WebRTCConnectionManager: Sendspin data channel state: Connecting
1775470497807 Debug/PeerConnectionWrapper[Android]: SDP offer created and set as local description
1775470497808 Debug/WebRTCConnectionManager: Sending SDP offer
1775470497811 Debug/SignalingClient: Sending signaling message: offer
1775470497882 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775470497884 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775470497884 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775470497885 Debug/SignalingClient: Sending signaling message: ice-candidate
1775470497885 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775470497885 Debug/SignalingClient: Sending signaling message: ice-candidate
1775470497916 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775470497917 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775470497918 Debug/SignalingClient: Sending signaling message: ice-candidate
1775470499236 Info/WebRTCConnectionManager: Disconnecting WebRTC connection
1775470499237 Info/MainDataSource: SessionState changed: Connecting
1775470499237 Info/MainDataSource: Connecting - stopping Sendspin
1775470499238 Info/SendspinClientFactory: Destroying shared audio pipeline
1775470499239 Debug/WebRTCTransport: Created new WebRTC manager [139806653]
1775470499239 Info/DataChannelWrapper[Android]: Closing data channel
1775470499239 Info/WebRTCConnectionManager: Starting WebRTC connection
1775470499239 Info/SignalingClient: Connecting to signaling server: [REDACTED_URL]
1775470499238 Error/WebRTCConnectionManager: Error collecting data channels
kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@713a2b9

1775470499238 Error/WebRTCConnectionManager: Error collecting ICE candidates
kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@306d6fe

1775470499239 Error/WebRTCConnectionManager: Error monitoring data channel state
kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@5df1e5f

1775470499239 Error/WebRTCConnectionManager: Error receiving messages from data channel
kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@ac2b175

1775470499239 Error/WebRTCConnectionManager: Error collecting connection state
kotlinx.coroutines.JobCancellationException: StandaloneCoroutine was cancelled; job=StandaloneCoroutine{Cancelling}@281417b

1775470499243 Info/DataChannelWrapper[Android]: Closing data channel
1775470499243 Info/PeerConnectionWrapper[Android]: Closing peer connection
1775470499244 Error/PeerConnectionWrapper[Android]: Data channel flow failed
kotlinx.coroutines.JobCancellationException: Job was cancelled; job=SupervisorJobImpl{Cancelling}@3587a57

1775470499244 Debug/WebRTCConnectionManager: Sendspin data channel state: Closed
1775470499244 Error/PeerConnectionWrapper[Android]: ICE candidate flow failed
kotlinx.coroutines.JobCancellationException: Job was cancelled; job=SupervisorJobImpl{Cancelling}@3587a57

1775470499244 Error/PeerConnectionWrapper[Android]: Connection state flow failed
kotlinx.coroutines.JobCancellationException: Job was cancelled; job=SupervisorJobImpl{Cancelling}@3587a57

1775470499249 Info/SignalingClient: Disconnecting from signaling server
1775470499250 Info/SignalingClient: Disconnected from signaling server
1775470499384 Info/SignalingClient: Connected to signaling server
1775470499385 Debug/WebRTCConnectionManager: Sending ConnectRequest message
1775470499386 Debug/SignalingClient: Sending signaling message: connect-request
1775470502984 Debug/SignalingClient: Received signaling message
1775470502988 Debug/SignalingClient: Parsed signaling message type: connected
1775470502989 Debug/WebRTCConnectionManager: Received signaling message: connected
1775470502990 Info/WebRTCConnectionManager: Connected. ICE servers: 4
1775470502991 Info/PeerConnectionWrapper[Android]: Initializing peer connection with 4 ICE servers
1775470503001 Debug/PeerConnectionWrapper[Android]: Peer connection initialized
1775470503002 Debug/WebRTCConnectionManager: Creating ma-api data channel
1775470503002 Debug/PeerConnectionWrapper[Android]: Creating data channel: ma-api (ordered=true, maxRetransmits=-1)
1775470503002 Debug/WebRTCConnectionManager: Peer connection state: new
1775470503006 Debug/WebRTCConnectionManager: Creating sendspin data channel
1775470503006 Debug/PeerConnectionWrapper[Android]: Creating data channel: sendspin (ordered=true, maxRetransmits=-1)
1775470503008 Debug/WebRTCConnectionManager: Creating SDP offer
1775470503008 Debug/PeerConnectionWrapper[Android]: Creating SDP offer
1775470503008 Debug/WebRTCConnectionManager: Sendspin data channel state: Connecting
1775470503020 Debug/PeerConnectionWrapper[Android]: SDP offer created and set as local description
1775470503020 Debug/WebRTCConnectionManager: Sending SDP offer
1775470503022 Debug/SignalingClient: Sending signaling message: offer
1775470503040 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775470503041 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775470503041 Debug/SignalingClient: Sending signaling message: ice-candidate
1775470503053 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775470503054 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775470503055 Debug/SignalingClient: Sending signaling message: ice-candidate
1775470503144 Debug/PeerConnectionWrapper[Android]: ICE candidate gathered
1775470503145 Debug/WebRTCConnectionManager: ICE candidate gathered, sending to signaling server
1775470503145 Debug/SignalingClient: Sending signaling message: ice-candidate
1775470508153 Debug/SignalingClient: Received signaling message
1775470508160 Debug/SignalingClient: Parsed signaling message type: answer
1775470508161 Debug/PeerConnectionWrapper[Android]: Connection state: connecting
1775470508162 Debug/WebRTCConnectionManager: Received signaling message: answer
1775470508162 Debug/WebRTCConnectionManager: Peer connection state: connecting
1775470508162 Info/WebRTCConnectionManager: Received SDP answer
1775470508162 Debug/PeerConnectionWrapper[Android]: Setting remote answer
1775470508168 Debug/PeerConnectionWrapper[Android]: Remote answer set successfully
1775470508168 Debug/WebRTCConnectionManager: Remote answer set successfully
1775470508208 Debug/PeerConnectionWrapper[Android]: Connection state: connected
1775470508209 Debug/WebRTCConnectionManager: Peer connection state: connected
1775470508224 Debug/WebRTCConnectionManager: Sendspin data channel state: Open
1775470508224 Info/WebRTCConnectionManager: Sendspin data channel ready for use
1775470508228 Info/MainDataSource: SessionState changed: WebRTC
1775470508230 Info/MainDataSource: Clearing all cached data
1775470508245 Info/MainDataSource: SessionState changed: WebRTC
1775470508246 Info/MainDataSource: Clearing all cached data
1775470508287 Debug/AuthVM: loadProviders() called, current providers count: 0
1775470508287 Debug/AuthVM: WebRTC connection - using builtin provider directly (skip API call)
1775470508288 Debug/AuthVM: SessionState changed: WebRTC
1775470508288 Debug/AuthVM: DataConnectionState: AwaitingAuth
1775470508288 Debug/AuthVM: AwaitingAuth - checking auth process state
1775470508289 Debug/AuthVM: Calling loadProviders()
1775470508289 Debug/AuthVM: loadProviders() called, current providers count: 1
1775470508383 Debug/AuthVM: SessionState changed: WebRTC
1775470508383 Debug/AuthVM: DataConnectionState: AwaitingAuth
1775470508383 Debug/AuthVM: AwaitingAuth - checking auth process state
1775470508383 Debug/AuthVM: Calling loadProviders()
1775470508383 Debug/AuthVM: loadProviders() called, current providers count: 1
1775470508384 Info/MainDataSource: SessionState changed: WebRTC
1775470508384 Info/MainDataSource: Clearing all cached data
1775470508446 Debug/ServiceClient: Saved token for server
1775470508447 Debug/AuthVM: SessionState changed: WebRTC
1775470508447 Debug/AuthVM: DataConnectionState: Authenticated
1775470508447 Info/MainDataSource: SessionState changed: WebRTC
1775470508450 Warn/MainDataSource: Cannot create Sendspin client: Sendspin disabled in settings
1775470508451 Info/MainDataSource: Updating players and queues
1775470509061 Info/MainMediaPlaybackService: Registered audio device callback for routing change detection
1775470509147 Info/MainMediaPlaybackService: Unregistered audio device callback
1775470510613 Info/: Player updated: Player(id=up0005cd34c87d, name=Denon RCD-N8, provider=universal_player, type=PLAYER, shouldBeShown=true, canSetVolume=true, volumeLevel=6.0, volumeControl=media_player.denon_rcd_n8, volumeMuted=false, canMute=true, queueId=up0005cd34c87d, isPlaying=true, isAnnouncing=false, canGroupWith=[], groupMembers=[], staticGroupMembers=[], groupVolume=6.0)
1775470529381 Debug/SignalingClient: Received signaling message
1775470529382 Debug/SignalingClient: Parsed signaling message type: ping
1775470529382 Debug/SignalingClient: Responding to signaling ping with pong
1775470529383 Debug/SignalingClient: Sending signaling message: pong
```

Okay, it crashes again when you change songs. 
```
1775470882030 Debug/WebRTCTransport: Created new WebRTC manager [96656095]
1775470882031 Info/WebRTCConnectionManager: Starting WebRTC connection
1775470882032 Info/SignalingClient: Connecting to signaling server: [REDACTED_URL]
1775470882056 Info/MainDataSource: SessionState changed: ByUser
1775470882056 Info/MainDataSource: Disconnected by user - clearing all data
1775470882057 Info/SendspinClientFactory: Destroying shared audio pipeline
1775470882057 Info/MainDataSource: Clearing all cached data

=== CRASH ===
java.lang.RuntimeException: Unable to create service io.music_assistant.client.services.MainMediaPlaybackService
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:5637)
	at android.app.ActivityThread.-$$Nest$mhandleCreateService(ActivityThread.java:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2796)
	at android.os.Handler.dispatchMessageImpl(Handler.java:142)
	at android.os.Handler.dispatchMessage(Handler.java:125)
	at android.os.Looper.loopOnce(Looper.java:292)
	at android.os.Looper.loop(Looper.java:393)
	at android.app.ActivityThread.main(ActivityThread.java:9486)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:575)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:939)
Caused by: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed: service io.music_assistant.client/.services.MainMediaPlaybackService
	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:54)
	at android.app.ForegroundServiceStartNotAllowedException$1.createFromParcel(ForegroundServiceStartNotAllowedException.java:50)
	at android.os.Parcel.readParcelableInternal(Parcel.java:5271)
	at android.os.Parcel.readParcelable(Parcel.java:5252)
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3378)
	at android.os.Parcel.createException(Parcel.java:3367)
	at android.os.Parcel.readException(Parcel.java:3350)
	at android.os.Parcel.readException(Parcel.java:3292)
	at android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:7511)
	at android.app.Service.startForeground(Service.java:775)
	at io.music_assistant.client.services.MainMediaPlaybackService.onCreate(MainMediaPlaybackService.kt:143)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:5626)
	... 10 more

```

--- a/androidApp/src/main/kotlin/io/music_assistant/client/services/AndroidAutoPlaybackService.kt
+++ b/androidApp/src/main/kotlin/io/music_assistant/client/services/AndroidAutoPlaybackService.kt
@@ -18,6 +18,7 @@ import coil3.ImageLoader
 import coil3.request.CachePolicy
 import coil3.request.ImageRequest
 import coil3.request.SuccessResult
+import coil3.request.allowHardware
 import io.music_assistant.client.R
 import io.music_assistant.client.auto.AutoLibrary
 import io.music_assistant.client.auto.MediaIds
@@ -378,6 +379,7 @@ class AndroidAutoPlaybackService : MediaBrowserServiceCompat() {
             ((imageLoader.execute(
                 ImageRequest.Builder(this@AndroidAutoPlaybackService)
                     .data(it)
+                    .allowHardware(false)
                     .memoryCachePolicy(CachePolicy.ENABLED)
                     .memoryCacheKey(it)
                     .build()

--- a/androidApp/src/main/kotlin/io/music_assistant/client/services/MainMediaPlaybackService.kt
+++ b/androidApp/src/main/kotlin/io/music_assistant/client/services/MainMediaPlaybackService.kt
@@ -22,6 +22,7 @@ import coil3.ImageLoader
 import coil3.request.CachePolicy
 import coil3.request.ImageRequest
 import coil3.request.SuccessResult
+import coil3.request.allowHardware
 import io.music_assistant.client.data.MainDataSource
 import io.music_assistant.client.ui.compose.common.DataState
 import io.music_assistant.client.ui.compose.common.action.PlayerAction
@@ -57,6 +58,7 @@ class MainMediaPlaybackService : MediaBrowserServiceCompat() {
     private lateinit var imageLoader: ImageLoader
     private lateinit var audioManager: AudioManager
     private var wifiLock: WifiManager.WifiLock? = null
+    private var fullyInitialized = false
 
     private val dataSource: MainDataSource by inject()
     private val sharedSession: SharedMediaSessionManager by inject()
@@ -69,7 +71,7 @@ class MainMediaPlaybackService : MediaBrowserServiceCompat() {
     private val activePlayerIndex = MutableStateFlow(-1)
     private val currentPlayerData =
         combine(players, activePlayerIndex) { players, index ->
-            // if some player is playing and we still have no valid index, show playing player
+            // if some player is playing, and we still have no valid index, show playing player
             if (index < 0 && players.isNotEmpty()) {
                 activePlayerIndex.update { max(0, players.indexOfFirst { it.player.isPlaying }) }
             }
@@ -142,14 +144,19 @@ class MainMediaPlaybackService : MediaBrowserServiceCompat() {
         )
         sessionToken = token
         mediaNotificationManager = MediaNotificationManager(this, token)
-        startForeground(
-            MediaNotificationManager.NOTIFICATION_ID,
-            mediaNotificationManager.createNotification(null)
-        )
-        imageLoader = ImageLoader(this)
-
-        // Register audio device callback to detect when external outputs disconnect
         audioManager = getSystemService(AUDIO_SERVICE) as AudioManager
+        imageLoader = ImageLoader(this)
+        try {
+            startForeground(
+                MediaNotificationManager.NOTIFICATION_ID,
+                mediaNotificationManager.createNotification(null)
+            )
+        } catch (e: IllegalStateException) {
+            logger.w(e) { "Cannot start foreground service from background, stopping" }
+            stopSelf()
+            return
+        }
+
         audioManager.registerAudioDeviceCallback(audioDeviceCallback, null)
         logger.i { "Registered audio device callback for routing change detection" }
 
@@ -171,6 +178,7 @@ class MainMediaPlaybackService : MediaBrowserServiceCompat() {
         acquireWifiLock()
         dataSource.apiClient.onPlaybackActive()
         registerNotificationDismissReceiver()
+        fullyInitialized = true
     }
 
     @Suppress("DEPRECATION")
@@ -295,11 +303,13 @@ class MainMediaPlaybackService : MediaBrowserServiceCompat() {
     }
 
     override fun onDestroy() {
-        releaseWifiLock()
-        unregisterReceiver(notificationDismissReceiver)
-        audioManager.unregisterAudioDeviceCallback(audioDeviceCallback)
-        logger.i { "Unregistered audio device callback" }
-        dataSource.apiClient.onPlaybackInactive()
+        if (fullyInitialized) {
+            releaseWifiLock()
+            unregisterReceiver(notificationDismissReceiver)
+            audioManager.unregisterAudioDeviceCallback(audioDeviceCallback)
+            logger.i { "Unregistered audio device callback" }
+            dataSource.apiClient.onPlaybackInactive()
+        }
         sharedSession.release(isAutoService = false)
         scope.cancel()
         super.onDestroy()
@@ -310,6 +320,7 @@ class MainMediaPlaybackService : MediaBrowserServiceCompat() {
             ((imageLoader.execute(
                 ImageRequest.Builder(this@MainMediaPlaybackService)
                     .data(it)
+                    .allowHardware(false)
                     .memoryCachePolicy(CachePolicy.ENABLED)
                     .memoryCacheKey(it)
                     .build()

